ksqldb-execution/src/main/java/io/confluent/ksql/execution/plan/StreamSource.java

```diff
@@ -41,14 +42,16 @@ public StreamSource(
       @JsonProperty(value = "topicName", required = true) final String topicName,
       @JsonProperty(value = "formats", required = true) final Formats formats,
       @JsonProperty("timestampColumn") final Optional<TimestampColumn> timestampColumn,
-      @JsonProperty(value = "sourceSchema", required = true) final LogicalSchema sourceSchema
+      @JsonProperty(value = "sourceSchema", required = true) final LogicalSchema sourceSchema,
+      @JsonProperty("pseudoColumnVersion") final OptionalInt pseudoColumnVersion
   ) {
     super(
         props,
         topicName,
         formats,
         timestampColumn,
-        sourceSchema
+        sourceSchema,
+        pseudoColumnVersion.orElse(0)
```

Comment:
Let's pull the value 0 out into a constant, e.g., `LEGACY_PSEUDOCOLUMN_VERSION_NUMBER`.